elasticsearch/_async/client/inference.py

```diff
@@ -234,6 +240,117 @@ async def get(
             path_parts=__path_parts,
         )
 
+    @_rewrite_parameters(
+        body_fields=("input", "query", "task_settings"),
+    )
+    @_stability_warning(
+        Stability.DEPRECATED,
+        version="8.18.0",
+        message="inference.inference() is deprecated in favor of provider-specific APIs such as inference.put_elasticsearch() or inference.put_hugging_face()",
+    )
+    async def inference(
+        self,
+        *,
+        inference_id: str,
+        input: t.Optional[t.Union[str, t.Sequence[str]]] = None,
+        task_type: t.Optional[
+            t.Union[
+                str,
+                t.Literal[
+                    "chat_completion",
```

Comment:
This is probably okay, but just noting that `chat_completion` is not in the list of task types in the current (8.17) client.

Reply:
Right, I wondered about it. Not including it encourages using the new way (`post_eis_chat_completion`, I guess) but having it allows users to keep using it for now.

Reply:
It's fine to keep it if it is a legal value. I only pointed it out in case it was a mistake.